### ISSUE TITLE
[Vega 3] Bin transform no longer support `field` and require extent?

### ISSUE THREAD

**Issue by kanitw** [vega/vega#663]
In Vega 2, `bin` transform can take `field` as input and determine `extent` (`[min,max]`) from the field.  

It seems like in Vega 3, this `field` property is no longer supported.  
I'm not sure if this is intended, if so, we should include this in the PORTING_GUIDE.

Note: another related things for bin is that `min` and `max` now becomes `extent`.  
(I also wonder if the same happens for any other transforms.)

**Comment by kanitw** [comment]
Note that if we no longer support `field`, I guess we have to create new data source to calculate min and max to feed the bin?  (If so, this seems quite more complicated from a user's perspective.)

**Comment by jheer** [comment]
You can add an `extent` transform and bind the result to a signal to get the min/max. For example, in the histogram.vg.json example you could remove the signal named `"extent"` and replace it with the following transform in the first data set definition:
```
"transform": [
        {
          "type": "extent",
          "field": "u",
          "signal": "extent"
        }
      ]
```
Currently, for internal reasons this extent transform can not be defined in the _same_ set of transforms as the `bin` transform. This is something I'd like to potentially fix before the 3.0 release, but hopefully shouldn't impede Vega-Lite 2.0 efforts for the moment.

I've opened two issues on vega/vega-dataflow to track this:
- vega/vega-dataflow#12
- vega/vega-dataflow#13
